### PR TITLE
Authentication, reconnects, mac hardware encoding, and fix leaky subscriptions

### PR DESCRIPTION
**Pull request by billylindeman**
#### Description
- Implements Authentication for websocket and rooms (using HMAC JWT)
- Adds config options to kubernetes chart for remb/pli
- Handles websocket reconnects
- Fixes issue with subscriptions being leaked when users disconnect
  -This causes a terrible cycle of write amplification for users with a flaky connection.  UDP Packets continue to be sent to the disconnected peers, and upon reconnection more subscriptions are added. 

cc 
@dreamerns @timsu

### CODE REVIEW

File: pkg/rtc/router.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ type RouterConfig struct {
 type Router struct {
 	pub           transport.Transport
 	subs          map[proto.MID]transport.Transport
+	userSubs      map[proto.UID]proto.MID
```
Comment (tarrencev) [inline pkg/rtc/router.go:37]: what is the difference between `userSubs` and `subs`?

**Review by tarrencev — APPROVED** [review]

File: pkg/rtc/router.go (hunk 1 above)
Comment (billylindeman) [reply pkg/rtc/router.go:37]: as far as I understand it this is tracking which UID is subscribed to this router's MID's so when that user disconnects we can clean the subscriptions up. @dreamerns is that correct?

File: configs/biz.toml [hunk 2]
```diff
@@ -21,8 +21,18 @@ host = "0.0.0.0"
...
+path = "/ws"
 allow_disconnected = false
 
+[signal.auth_connection]
+enabled = true 
+key_type = "HMAC"  # this selects the Signing method https://godoc.org/github.com/dgrijalva/jwt-go#SigningMethod
+key = "1q2dGu5pzikcrECJgW3ADfXX3EsmoD99SYvSVCpDsJrAqxou5tUNbHPvkEFI4bTS"
+
+[signal.auth_room]
+enabled = true
```
Comment (timsu) [inline configs/biz.toml:33]: shouldn't these be off by default? otherwise this won't work out of the box with sample clients

Comment (billylindeman) [reply configs/biz.toml:33]: 👍  yep

File: pkg/rtc/router.go (hunk 1 above)
Comment (dreamerns) [reply pkg/rtc/router.go:37]: Yes. Once user leaves the room, we would clean up all its streams as well as subscriptions to those streams. However, the subscriptions user had to other streams would still be there, causing RTP data being sent to the user even when he is no longer present in the room. This helps track user MID, so we can clean up those subscriptions once user leaves the room.

**Pull request merged**